frontend/pages/dashboard.vue

```diff
@@ -1,25 +1,238 @@
+let searchValue = ref<String>('');
+let currentJourneys = ref<Journey[]>([]);
+
+currentJourneys.value = store.journeys;
+
+const menu = ref();
+const items = ref([
+  {
+    label: t.value('dashboard.sort.name'),
+    icon: 'pi pi-book',
+    items: [
+      {
+        label: t.value('dashboard.sort.ascending'),
+        icon: 'pi pi-sort-alpha-up',
+        command: () => {
+          sortJourneys('name-asc');
+        }
+      },
+      {
+        label: t.value('dashboard.sort.descending'),
+        icon: 'pi pi-sort-alpha-down',
+        command: () => {
+          sortJourneys('name-desc');
+        }
+      },
+    ]
+  },
+  {
+    label: t.value('dashboard.sort.startdate'),
+    icon: 'pi pi-calendar',
+    items: [
+      {
+        label: t.value('dashboard.sort.oldestToNewest'),
+        icon: 'pi pi-sort-amount-up',
+        command: () => {
+          sortJourneys('startdate-asc');
+        }
+      },
+      {
+        label: t.value('dashboard.sort.newestToOldest'),
+        icon: 'pi pi-sort-amount-down',
+        command: () => {
+          sortJourneys('startdate-desc');
+        }
+      },
+    ]
+  },
+  {
+    label: t.value('dashboard.sort.destination'),
+    icon: 'pi pi-map-marker',
+    items: [
+      {
+        label: t.value('dashboard.sort.ascending'),
+        icon: 'pi pi-sort-alpha-up',
+        command: () => {
+          sortJourneys('destination-asc');
+        }
+      },
+      {
+        label: t.value('dashboard.sort.descending'),
+        icon: 'pi pi-sort-alpha-down',
+        command: () => {
+          sortJourneys('destination-desc');
+        }
+      },
+    ]
+  },
+
+]);
+
+const toggle = (event: Event) => {
+  menu.value.toggle(event);
+};
+
+
+async function fetchJourneys() {
+  const client = useSanctumClient();
+  await client("/api/journey", {
+    method: "GET",
+    async onResponse({ response }) {
+      journeys.value = response._data;
+      currentJourneys.value = response._data;
+      store.setJourneys(response._data);
+    }
+  });
+
+}
+
+
+async function searchJourneys() {
+  const data: Journey[] = journeys.value;
+  const results: Journey[] = data.filter((obj: Journey) => {
+    return obj.name.toLowerCase().includes(searchValue.value.toLowerCase()) ||
+      obj.destination.toLowerCase().includes(searchValue.value.toLowerCase());
+  });
+
+  currentJourneys.value = results;
+}
+
+function sortJourneys(sortKey: String) {
+  currentJourneys.value.sort((a: Journey, b: Journey) => {
+    if (sortKey === 'name-asc') return b.name.localeCompare(a.name);
+    if (sortKey === 'name-desc') return a.name.localeCompare(b.name);
+    if (sortKey === 'startdate-asc') return a.from.localeCompare(b.from);
+    if (sortKey === 'startdate-desc') return b.from.localeCompare(a.from);
+    if (sortKey === 'destination-asc') return b.destination.localeCompare(a.destination);
+    if (sortKey === 'destination-desc') return a.destination.localeCompare(b.destination);
+    return 0;
+  });
+}
```

Comment:
Maybe a switch statement would be better suited for this purpose? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch